### PR TITLE
New comparer interface

### PR DESCRIPTION
**Pull request by asi1024**

### CODE REVIEW

File: pytorch_pfn_extras/utils/comparer.py [hunk 1]
```diff
@@ -296,3 +300,197 @@ def _add_target(self, handle, models, outputs):
...
+
+        Args:
+            loaders (dict of loaders):
+                Data loaders used as input for each engine.
+        """
+        n_workers = len(self._engines)
+        self._barrier = threading.Barrier(n_workers)
+        self._semaphore = threading.Semaphore(
+            n_workers if self._concurrency is None else self._concurrency)
+        with concurrent.futures.ThreadPoolExecutor(max_workers=n_workers) as executor:
```
Comment (kmaehashi) [inline pytorch_pfn_extras/utils/comparer.py:488]: Could you add a path to run without ThreadPoolExecutror when `self._concurrency == None`?

**Review by kmaehashi — CHANGES_REQUESTED** [review]
Thanks~! :+1:

File: pytorch_pfn_extras/utils/comparer.py [hunk 2]
```diff
@@ -296,3 +300,197 @@ def _add_target(self, handle, models, outputs):
             self._preprocess_keys(sdict)
         self.targets[handle.name] = {
             key: sdict[key] for key in self._preprocessed_keys}
+
+
+# New comparer interface
+
+def _filter(
+        keys: Union[bool, str, Sequence[str]],
+        get_dict: Callable[[], Dict[str, Any]],
```
[the remark this answers is not in the document]
Comment (asi1024) [reply pytorch_pfn_extras/utils/comparer.py:309]: Because I don't want to call `model.state_dict()` when model parameters are not compared.

**Comment by asi1024** [comment]
PTAL.
/test

**Review by kmaehashi — APPROVED** [review]

**Pull request merged**